### PR TITLE
feature: create package eslint config to tignum fronted proyects 

### PR DESCRIPTION
**Pull request by masedri**
# Ticket https://tignum.atlassian.net/browse/WEBDEV-2588

## What?

Fill out a short description about the ticket.

## Why?

Fill out why this feature is needed. Kind of the same stuff that is usually mentioned in the jira ticket.

## How?

Explain in short terms what you did here, what your solution/implementation is doing.

## Testing?

Tell us the test cases you did in order to prove that this feature is working.

### CODE REVIEW

File: github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Ticket https://tignum.atlassian.net/browse/WEBDEV
```
Comment (dgrabla) [inline github/pull_request_template.md:1]: Please use the template we agreed upon https://github.com/TIGNUM/qot-web-application/blob/develop/.github/pull_request_template.md

File: github/workflows/release-package.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    types: [created]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 12
```
Comment (dgrabla) [inline github/workflows/release-package.yml:14]: use 16, or LTS

File: package.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    "url": "https://github.com/TIGNUM/eslint-config-tignum.git"
+  },
+  "publishConfig": {
+    "registry": "https://npm.pkg.github.com"
+  },
+  "author": "m.rosero@tignum.com",
+  "license": "MIT",
+  "homepage": "https://github.com/TIGNUM/eslint-config-tignum#readme",
+  "devDependencies": {
+    // "eslint-plugin-sonarjs": "^0.13.0",
```
Comment (dgrabla) [inline package.json:17]: This looks weird, does it work for you to have comments on a json file?

File: rules/vue.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  ignorePatterns: ['package.json', 'dist/'],
+  extends: [
+    'airbnb',
+    'eslint:recommended',
+    'plugin:vue/vue3-recommended',
+    'prettier',
+    'plugin:import/errors'
+  ],
+  rules: {
+    'vue/no-v-html': 0
```
Comment (dgrabla) [inline rules/vue.js:43]: This is missing a chunk of the rules we need for the memberhub. I however don't know if we use the default values and that's why we don't need them here

https://github.com/TIGNUM/qot-web-application/blob/develop/.eslintrc.js

image

File: rules/cypress.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  plugins: ['cypress'],
+  extends: ['plugin:cypress/recommended'],
+  rules: {
+    'cypress/no-unnecessary-waiting': 'warn'
```
Comment (dgrabla) [inline rules/cypress.js:5]: This seems to be new, right?

**Review by dgrabla — CHANGES_REQUESTED** [review]
Thanks @masedri !

Here is some early feedback:

I'm missing here the 'unscoped' set of rules we have on the https://github.com/TIGNUM/qot-web-application/blob/develop/.eslintrc.js.

Remember ideally the first version of the package should be able to be imported on the memberhub WITHOUT any new error/warning. Then we can import the package in all the other projects to assure coherence, and LATER we can discuss introducing new code style.

File: package.json (hunk 3 above)
Comment (masedri) [reply package.json:17]: You can enable it in vscod, it's just a reminder for the readme

File: README.md [hunk 6]
```diff
@@ -1 +1,39 @@
...
+
+1. Install package
+```
+pnpm add -D @tignum/eslint-config
+```
+
+2. Create config file .eslintrc and add
+```
+{
+    "extends": "@tignum"
```
[the remark this answers is not in the document]
Comment (masedri) [reply README.md:17]: You can use only scope of the organization too. https://eslint.org/docs/developer-guide/shareable-configs#npm-scoped-modules

File: rules/vue.js (hunk 4 above)
Comment (masedri) [reply rules/vue.js:43]: This is a WIP, I'm working in it ^^

File: rules/cypress.js (hunk 5 above)
Comment (masedri) [reply rules/cypress.js:5]: yes, this a rule is to when use wait in cypress https://github.com/cypress-io/eslint-plugin-cypress#disable-rules

**Comment by masedri** [comment]
> Thanks @masedri !
> 
> Here is some early feedback:
> 
> I'm missing here the 'unscoped' set of rules we have on the https://github.com/TIGNUM/qot-web-application/blob/develop/.eslintrc.js.
> 
> Remember ideally the first version of the package should be able to be imported on the memberhub WITHOUT any new error/warning. Then we can import the package in all the other projects to assure coherence, and LATER we can discuss introducing new code style.

Hi @dgrabla 

ty for your feedback! I will working in your comments! 👍 

I'dont know if a good idea add rules of qot web, I'm trying to create modular config linter for different options and a base linter with the necessary config and good practice, then we can extends this config and add-remove rules need in qot .

I don't understand very well the rules of linter qot or if we follow a standard, it's a bit crazy so many rules, right? I think I could mess up the @tignum/eslint-config package

We should build on a standard linter, for example, recommended, standard or airbnb and then add or remove rules.

Could we discuss it in a call? maybe I don't have enough context, thanks!

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (dgrabla) [reply README.md:17]: :+1:

File: package.json (hunk 3 above)
Comment (dgrabla) [reply package.json:17]: It is not only the editor. No package manager likes them
image
image
image

File: rules/cypress.js (hunk 5 above)
Comment (dgrabla) [reply rules/cypress.js:5]: Okidoki. It is fine to define rules as long as it does require changes on the cypress tests when we introduce this package on the memberhub

**Pull request merged**